subsys/storage/flash_map/flash_map.c

```diff
@@ -281,8 +281,8 @@ int flash_area_check_int_sha256(const struct flash_area *fa,
 	int pos;
 	int rc;
 
-	if (!fa || !fac || !fac->match || !fac->rbuf ||
-	    fac->clen <= 0 || fac->off < 0 || fac->rblen <= 0) {
+	if (fa == NULL || fac == NULL || fac->match == NULL ||
+	    fac->rbuf == NULL || fac->clen == 0 || fac->rblen == 0) {
```

Comment:
check on fac->off is missing, why?

Reply:
maybe it is too offensive? :-D

Reply:
Still, I think @de-nordic needs to address the question :)

Reply:
(Hah!) And the problem this PR is addressing is that `off` is an unsigned type, so it can never be negative.

Reply:
OK thanks for clarifying this @pabigot . This was not intuitive, since other similar checks in this line have been reworked from inequality checks (<= 0) to equality (==) so I was thinking this should be the same and was just overlooked (possibly because it was too offensive :) )

Reply:
@ioannisg off is very positive when negative.

Reply:
Hi folks, It was not intentional flash_area_check -> offset be offensive.

Reply:
Maybe this can be refactored, then :)

Reply:
Maybe leave it, it is funny :), it is one of these things that give you smile when you scan the code :)